frontend/src/metabase/query_builder/actions/sharing.ts

```diff
@@ -0,0 +1,44 @@
+import { createAction } from "redux-actions";
+import type { Card, CardId } from "metabase-types/api";
+import type { EmbedOptions } from "metabase-types/store";
+import { CardApi } from "metabase/services";
+
+type CardIdPayload = {
+  id: CardId;
+};
+
+export const CREATE_PUBLIC_LINK = "metabase/card/CREATE_PUBLIC_LINK";
+
+export const createPublicLink = createAction(
+  CREATE_PUBLIC_LINK,
+  ({
+    id,
+  }: Card): Promise<
+    CardIdPayload & {
```

Comment:
I'd inline `id: CardId` instead. When I see `Type & { property: Type }` I'd expect `Type` to have a handful of keys otherwise, it's easier to read the return type when inlining everything since it only has 2 properties in this case.